src/main/java/ryze/Ryze.java

```diff
@@ -0,0 +1,380 @@
+        return lastSize;
+    }
+
+    private static void initialiseData() {
+        Path dirPath = Paths.get("data");
+        Path filePath = dirPath.resolve("ryze.txt");
+        String filePathString = filePath.toString();
+        File file = new File(filePathString);
+        System.out.println(filePathString);
+
+        // Check if directory exists, if not, create it
+        if (!Files.exists(dirPath)) {
+            try {
+                Files.createDirectories(dirPath);
+            } catch (IOException e) {
+                System.out.println("Error creating directory: " + e.getMessage());
+                return;
+            }
+        }
+
+        // Check if file exists, if not, create it
+        if (!file.exists()) {
+            System.out.println("Data for Ryze not found! Welcome to Ryze new User!");
+            try {
+                Files.createFile(filePath); // Creates the file if it doesn't exist
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        } else {
+            try {
+                readFileContents(filePathString);
+            } catch (IOException e) {
+                System.out.println("Error reading file: " + e.getMessage());
+            }
+        }
+    }
+
+    private static void appendToFile(String textToAppend) throws IOException {
+        Path dirPath = Paths.get("data");
+        Path filePath = dirPath.resolve("ryze.txt");
+        String filePathString = filePath.toString();
+        FileWriter fw = new FileWriter(filePathString, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    private static void readFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            parseRyzeTxt(s.nextLine());
+        }
+    }
+
+    private static void parseRyzeTxt(String dataEntry) {
+        String taskType = dataEntry.split("~")[0];
+        switch (taskType) {
+        case TODO:
+            parseTodo(dataEntry);
+            break;
+        case DEADLINE:
+            parseDeadline(dataEntry);
+            break;
+        case EVENT:
+            parseEvent(dataEntry);
+            break;
+        default:
+            System.out.println("Invalid task type: " + taskType);
+        }
+    }
+
+    private static void parseEvent(String dataEntry) {
+        String description = dataEntry.split("~")[2];
+        String from = dataEntry.split("~")[3];
+        String to = dataEntry.split("~")[4];
+        Task newEvent = new Event(description,from,to);
+        boolean isDone = Integer.parseInt(dataEntry.split("~")[1]) != 0;
+        if(isDone){
+            newEvent.markAsDone();
+        }
+        listOfChatHistory.add(newEvent);
+    }
```

Comment:
You may consider extracting this part further and passing in parameters to differentiate event/deadline etc.